bindings/updater/nodejs/src/lib.rs

```diff
@@ -178,94 +179,196 @@ impl Update {
     }
 }
 
+type TaskCallbackFunction<T> = Option<ThreadsafeFunction<T, ErrorStrategy::Fatal>>;
+
+pub struct DownloadTask {
+    update: cargo_packager_updater::Update,
+    on_chunk: TaskCallbackFunction<(u32, Option<u32>)>,
+    on_download_finished: TaskCallbackFunction<()>,
+}
+
+impl DownloadTask {
+    pub fn create(
+        update: &Update,
+        on_chunk: TaskCallbackFunction<(u32, Option<u32>)>,
+        on_download_finished: TaskCallbackFunction<()>,
+    ) -> Result<Self> {
+        Ok(Self {
+            update: update.create_update()?,
+            on_chunk,
+            on_download_finished,
+        })
+    }
+}
+
+impl Task for DownloadTask {
```

Comment:
could use a `#[derive(napi_derive::napi)] impl Task for T` instead of the `ts_return_type` hack but it's still not perfect for the JsArrayBuffer type, so doesn't matter